src/engine/drpc_client.c

```diff
@@ -171,6 +174,26 @@ drpc_notify_ready(void)
 	if (rc != 0)
 		goto out_uri;
 
+	nr_sec_uris = crt_get_nr_secondary_providers();
+	if (nr_sec_uris > 0) {
+		D_ALLOC_ARRAY(sec_uris, nr_sec_uris);
+		if (sec_uris == NULL)
```

Comment:
i think rc has to be set to -DER_NOMEM here

Reply:
fixed